### ISSUE TITLE
simple constructor `ReactionSystem(rxns)`

### ISSUE THREAD

**Issue by anandijain** [SciML/Catalyst.jl#425]
maybe there is not an analog but MTK has a really nice `ODESystem(eqs)` constructor that finds all the parameters, states, etc.

I think it'd be nice to have this for ReactionSystem.

rough sketch
```julia 
species(rxn::Reaction) = unique(reduce(vcat, (rxn.substrates, rxn.products)))
species(rxns::Vector{<:Reaction}) = unique(reduce(vcat, map(rxnspecies, rxns)))

ReactionSystem(rxn::Reaction; kwargs...) = ReactionSystem(Reaction[rxn]; kwargs...)

function ReactionSystem(rxns::Vector{<:Reaction}; kwargs...)
    ReactionSystem(rxns, Catalyst.DEFAULT_IV, species(rxns), []; kwargs...)
end
```

but I didn't know how to find all the parameters for a reaction. I think we disallow symbolic params in stoich coefficients, right? 

That means I'd just need to find them in the `rate` part, I think.

This isn't urgent but I bumped up against it so I wanted to document this

**Comment by isaacsas** [comment]
Hmm, parameters now have meta-data tagging them right? So I guess we could walk all reaction rate expressions and identify if each term is a parameter, and if not declare it a species? (i.e. we need to look at the `rx.rate`s too, not just the `rx.substrates` and `rx.products` -- this is actually an issue with the DSL that needs to get fixed too.) 

For now the stoichiometry only allows species, but I'd like to get more general stoichiometry in (the big hangup is the issues I mentioned in https://github.com/SciML/ModelingToolkit.jl/issues/1246).

**Comment by isaacsas** [comment]
This would be a great add though; not having to list the parameters, `iv` and species would be a nice improvement!

**Comment by isaacsas** [comment]
Once https://github.com/SciML/Catalyst.jl/pull/423 is merged I have a bit more to do and then I’ll make a new breaking release. Might be a week or so.

**Comment by anandijain** [comment]
is it breaking? i don't think so, right

**Comment by isaacsas** [comment]
This isn’t, but other unreleased changes are I think.
